### PR TITLE
Fix tests on arm64 & GH Actions

### PR DESCRIPTION
**Pull request by douglascamata**
Also on amd64, as a bonus. `make test` wasn't passing on a fresh instance of Codespaces before the changes.

I took the freedom to fix the Github Actions configuration that was programmed to run builds on pushes to `master`, but this repository uses `main` as default branch.

### CODE REVIEW

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go: [ '1.16.x']
+        go: ["1.16.x"]
```
Comment (bwplotka) [inline .github/workflows/go.yaml:37]: Should we update this?

**Review by bwplotka — APPROVED** [review]

File: .github/workflows/go.yaml (hunk 1 above)
Comment (douglascamata) [reply .github/workflows/go.yaml:37]: I'm happy to add this to my open source todo list. I can land it next week. :)

**Comment by douglascamata** [comment]
@bwplotka due to issues with the GH Actions being used to setup Docker in the macOS runner, I moved to a simpler alternative solution using [Colima](https://github.com/abiosoft/colima) and [Lima-VM](https://github.com/lima-vm/lima). Lima-VM is using QEMU with some sort of acceleration called HVF that makes it suitable for this task. It's the same mechanism that Rancher Desktop uses under the hood.

File: .github/workflows/go.yaml [hunk 2]
```diff
@@ -47,9 +47,11 @@ jobs:
         with:
           go-version: ${{ matrix.go }}
 
-      - name: Install Docker
+      - name: Install docker - MacOS
         if: runner.os == 'macOS'
-        uses: docker-practice/actions-setup-docker@master
+        run: |
+          brew install docker colima
```
Comment (douglascamata) [inline .github/workflows/go.yaml:53]: This installs only the Docker client CLI, not the Docker Desktop application.

**Review by bwplotka — APPROVED** [review]
💪🏽 

Thanks!

**Pull request merged**